chandra_aca/centroid_resid.py

```diff
@@ -0,0 +1,245 @@
+    return mult
+
+
+class CentroidResiduals(object):
+    centroid_source = None
+    att_source = None
+    ra = None
+    dec = None
+
+    def set_centroids(self, start, stop, source, slot, alg=8):
+        """
+        Get centroids from ``source`` between ``start`` and ``stop``.
+
+        One could also just set yags, yag_times, zags, zag_times
+        attributes directly.
+
+        :param start: start time
+        :param stop: stop time
+        :param source: 'ground' | 'obc'
+        :param slot: ACA slot
+        """
+        self.centroid_source = source
+        # Get centroids from Ska eng archive or mica L1 archive
+        # Might want to add filtering for track status here too
+        # Also need to include something for time offsets
+        # And warn to console if we hit multiple obsids
+        if source == 'ground':
+            acen_files = sorted(asp_l1.get_files(start=start, stop=stop, content=['ACACENT']))
+            acen = vstack([Table.read(f) for f in sorted(acen_files)], metadata_conflicts='silent')
+            ok = ((acen['slot'] == slot) & (acen['alg'] == alg) & (acen['status'] == 0)
+                  & (acen['time'] >= DateTime(start).secs) & (acen['time'] <= DateTime(stop).secs))
+            yags = np.array(acen[ok]['ang_y'] * 3600)
+            zags = np.array(acen[ok]['ang_z'] * 3600)
+            yag_times = np.array(acen[ok]['time'])
+            zag_times = np.array(acen[ok]['time'])
+        elif source == 'obc':
+            telem = fetch.Msidset(['AOACYAN{}'.format(slot), 'AOACZAN{}'.format(slot)], start, stop)
+            # Filter centroids for reasonble-ness
+            yok = telem['AOACYAN{}'.format(slot)].vals != -3276.8
+            zok = telem['AOACZAN{}'.format(slot)].vals != -3276.8
+            yags = telem['AOACYAN{}'.format(slot)].vals[yok]
+            yag_times = telem['AOACYAN{}'.format(slot)].times[yok]
+            zags = telem['AOACZAN{}'.format(slot)].vals[zok]
+            zag_times = telem['AOACZAN{}'.format(slot)].times[zok]
+            raise ValueError
+        self.yags = yags
+        self.yag_times = yag_times
+        self.zags = zags
+        self.zag_times = zag_times
+
+    def set_atts(self, start, stop, source):
+        """Get attitude solution quaternions from ``source`` between
+        ``start`` and ``stop``
+
+        One could also just set atts and att_times attributes directly.
+        """
+        self.att_source = source
+        # Get attitudes and times
+        if source == 'obc':
+            # Need to figure out missing data
+            telem = fetch.Msidset(['aoattqt*'], start, stop)
+            atts = np.vstack([telem['aoattqt{}'.format(idx)].vals
+                              for idx in [1, 2, 3, 4]]).transpose()
+            att_times = telem['aoattqt1'].times
+        if source == 'ground':
+            asol_files = sorted(asp_l1.get_files(start=start, stop=stop, revision='last',
```

Comment:
Not sure if this should be revision='last' or the default release.  Or if that should be an option to the method.